enhancements/bare-metal-fulfillment/README.md

```diff
@@ -0,0 +1,267 @@
+
+We will not substantially modify the existing cluster fulfillment workflow, although we believe bare metal fulfillment should
+eventually be used as the basis of bare metal management within cluster fulfillment.
+
+## Proposal
+
+The implementation of bare metal fulfillment relies on three concepts:
+
+* HostPool: A collection of requested bare metal machines, along with their desired network configuration.
+* Host: An ephemeral resource created when bare metal resources are acquired from the underlying bare metal management layer. They represent the assignment of a bare metal resource to a HostPool, and provide an endpoint for host-specific operations.
+* HostClass: Defines a set of bare metal properties, such as memory, CPU, and network interfaces. Each Host has a single HostClass attribute.
+
+At a high level, tenants will request a HostPool by specifying:
+
+* the number and HostClass of hosts
+* the network configuration to be applied to those hosts (note that this configuration applies to each individual host in the HostPool)
+
+The O-SAC solution will assign the requested hosts to the tenant and create matching Host resources for each host. The tenant
+can then perform needed operations upon the Host.
+
+We expect bare metal fulfillment to follow the same workflow used for cluster fulfillment; as such, we expect to update the
+following existing O-SAC components:
+
+* Fulfillment Service: Define the API for HostPools, Hosts, and HostClasses
+* Fulfillment CLI: Give the tenant access to the API
+* Cloudkit Operator: Manage and reconcile the Custom Resources for HostPools and Hosts
+* Cloudkit AAP: Use Ansible playbooks to perform the requested reconciliation operations of HostPools and Hosts by calling Bare Metal Management APIs.
+* Bare Metal Management: Manage the hardware; we will use ESI for now
+
+Much of the development work can simply follow the established patterns set by the cluster fulfillment workflow. However, the
+following topics merit further discussion in the "Implementation Details/Notes/Constraints" below:
+
+* ESI playbooks
+* Serial console support
+
+### Workflow Description
+
+#### Host Pool Creation
+
+1. The tenant uses the Fulfillment CLI to request a HostPool.
+2. The O-SAC solution fulfills the request and creates Host resources corresponding to the allocated hosts.
+
+#### Host Pool Network Update
+
+1. The tenant uses the Fulfillment CLI to update the network configuration of a HostPool.
+2. The O-SAC solution fulfills the request by updating the network configuration of the hosts in the HostPool.
+
+#### Host Operations
+
+1. The tenant uses the Fulfillment CLI to see their Hosts.
+2. The tenant uses the Fulfillment CLI to perform an operation upon the desired Host.
+
+#### Host Pool Deletion
+
+1. The tenant uses the Fulfillment CLI to delete a HostPool.
+2. The O-SAC solution fulfills the request and removes the Host resources corresponding to the hosts in the HostPool.
+
+### API Extensions
+
```

Comment:
Should we say how this fits into design, i.e., key features of hosts... are part of the API so operator can validate not changed

Reply:
I'm not entirely sure what you mean here - do you mean how this proposal fits into the generic O-SAC fulfillment model?